app/lib/actions/katello/repository/destroy.rb

```diff
@@ -70,11 +70,7 @@ def handle_acs_product_removal(repository)
           # Remove products from ACS's that contain no repositories which both
           # match the ACS content type and have a non-nil URL
           product = repository.product
-          repo_content_types = Set.new
```

Comment:
This slow logic also exists in `app/lib/actions/katello/repository/update.rb` when a user removes a product's url. Would you mind replacing the old loop with your new lookup there as well? There's a chance `plans ACS product removal when removing the last URL for a product` will break.

Reply:
Good catch, I updated the other file the same way.